Add Client connection to SignalR sever

bogse opened the pull request:
Add Client connection to the SignalR server.

Please notice Unit test passes only when the server is up.
I tried different approaches: instantiating both real connections and mock of HubConnection class, wrapper over HubConnection to expose the State property, all of them failing with HubConnection.State is readonly and cannot be set or 'Castle.Proxies.HubConnectionProxy' can not instantiate proxy of class errors.
Should we test the Connect() method of SignalR or just the methods that send messages?

Added ConsoleClient to the HuntTheWumpus.
Cleanup duplicate folder "System Testing".
Modified swagger to display Hub methods in the browser. Post methods from the browser are not working. (probably unnecessary modification).

bogse committed:
Add Client connection to SignalR sever

bogse committed:
Merge branch 'main' into feature/clientConnection

ioanateo commented on ConsoleClient/Connection/SignalRClient.cs:
```diff
@@ -1,12 +1,34 @@
+        public SignalRClient()
+        {
+            _playerHubConnection = new HubConnectionBuilder()
+                .WithUrl("https://localhost:7252/playerhub")
```
poti declara url ca si constanta in clasa asta
la fel si pentru url de mai jos

bogse replied:
am facut un baseUrl, ca apoi sa concatenez path-ul pentru fiecare hub in HubConnectionBuilder.
modific ulterior daca trebuia facuta o constanta diferita pentru fiecare url.

bogse committed:
Updated Connection class with baseUrl constant

E ok sa las o constanta baseUrl (doar cu protocol + domain), ca apoi sa fie concatenat path-ul fiecarui hub?
Sau fac 2 constante cu fiecare? 

Signed-off-by: Bogdan Ciucuras <90960071+bogse@users.noreply.github.com>

ileuca reviewed (commented):
Nu e un approach intocmai bun. Doua mai des folosite ar fi folosind un config file sau il adaugi ca environment variable in aplicatie.

1.Config File => cred ca poti folosi direct appsettings.json 

- adaugi constanta in fisier adaugi 
{
"baseUrl":"https://localhost:7252"
}

- ar trebui sa configurezi serviciile in Program.cs
services.Configure<AppSettings>(Configuration.GetSection("baseURL"));
("mai trebuie documentat ce trebuie adaugat si cum trebuie facut in Program.cs")

- creezi o clasa in care sa faci store la variabile gen 

class ConfigVars
{
public string baseURL;
}

- si ultimul pas parametrizezi constructorul clasei unde vrei sa il folosesti

In cazul de fata ar fi in genul:
```
public SignalRClient(IOptions<ConfigVars> configVars>
        {
            _playerHubConnection = new HubConnectionBuilder()
                .WithUrl(configVars.Value.baseURL + "/playerhub")
                .WithAutomaticReconnect()
                .Build();

            _caveHubConnection = new HubConnectionBuilder()
                .WithUrl(configVars.Value.baseURL + "/cavehub")
                .WithAutomaticReconnect()
                .Build();
        }
```

sau varianta 
2. EnvironmentVariables

Cand faci deploy la aplicatii vei avea mereu optiunea de a adauga environment variables.
baseURL= Environment.GetEnvironmentVariable("BaseURL");

Un mic sidenote: cand ai multe variabile va fi o durere sa le setezi cand faci deploy la app. motiv pentru care ar fi mai indicat sa folosesti un configFile. daca is putine merge pe environment variables.

Nu stiu care ar fi cel mai Clean si corect approach.

bogse committed:
Created appsettings.json for BaseUrl

bogse committed:
Add baseUrl in appsettings.json

bogse committed:
Refactor SignalRClient to get baseUrl configuration from ConfigVars class

bogse commented:
Am incercat cu prima varianta ( appsettings.json ), am creat unul nou in proiectul de ConsoleClient, asta sa evit sa fac referinta spre cel de GameServer (nu stiu daca e cel mai bun approach).

Dupa am recitit restul pasilor si mi-am dat seama ca nu facusem o clasa diferita sa se ocupe de config si am facut refactor.

Also, l-am pus in constructorul de client, dar nu stiu daca trebuia asa sau in Program.cs. Ii fac refactor daca este nevoie.
